core/services/workflows/v2/secrets.go

```diff
@@ -189,7 +195,8 @@ func (s *secretsFetcher) getSecrets(ctx context.Context, request *sdkpb.GetSecre
 		}
 
 		shares := resp.GetData().EncryptedDecryptionKeyShares[0].Shares
```

Comment:
@prashantkumar1982 I think this needs tweaking to -- you will get a number of `EncryptedDecryptionKeyShares` equal to the number of nodes in the DON and will need to find the shares whose public key matches your node's public key (the shares will be returned alongside the public key to facilitate this)